boards/DIM/Src/App/App_RotarySwitch.c

```diff
@@ -41,3 +42,14 @@ ExitCode App_RotarySwitch_GetSwitchPosition(
 
     return exit_code;
 }
+
+bool App_RotarySwitch_IsValid(const struct RotarySwitch *const rotary_switch)
+{
+    uint32_t position = rotary_switch->get_switch_position();
```

Comment:
nit - could make the function a bit more readable by simplifying
```suggestion
    return (rotary_switch->get_switch_position() < rotary_switch->num_switch_positions);
```